src/test/java/com/contentgrid/gateway/security/jwt/issuer/PropertiesBasedJwtClaimsSignerTest.java

```diff
@@ -0,0 +1,239 @@
+package com.contentgrid.gateway.security.jwt.issuer;
+
+import static com.contentgrid.gateway.security.jwt.issuer.CryptoTestUtils.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.contentgrid.gateway.security.jwt.issuer.PropertiesBasedJwtClaimsSigner.JwtClaimsSignerProperties;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSAlgorithm.Family;
+import com.nimbusds.jose.crypto.ECDSAVerifier;
+import com.nimbusds.jose.crypto.RSASSAVerifier;
+import com.nimbusds.jose.crypto.factories.DefaultJWSSignerFactory;
+import com.nimbusds.jose.crypto.factories.DefaultJWSVerifierFactory;
+import com.nimbusds.jose.jwk.AsymmetricJWK;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jwt.JWTClaimsSet;
+import java.nio.charset.StandardCharsets;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.Value;
+import lombok.experimental.Delegate;
+import org.junit.jupiter.api.Test;
+import org.springframework.core.io.support.ResourcePatternResolver;
+
+class PropertiesBasedJwtClaimsSignerTest {
+
+    static Random createDeterministicRandom() {
+        // This is a random that will always start from the same seed
+        return new Random(5);
```

Comment:
I'd bet sonar is going to love this ;-)

Reply:
1. This is a test class
2. Sonar doesn't seem to care about this one; it does complain about the other Random constructor in production code though.